--- grottdata.py	2022-08-20 23:31:20.000000000 +0200
+++ grottdata.patched.py	2022-08-20 23:31:37.000000000 +0200
@@ -374,8 +374,9 @@
 
         if conf.nomqtt != True:
             try: 
-                publish.single(conf.mqtttopic, payload=jsonmsg, qos=0, retain=True, hostname=conf.mqttip,port=conf.mqttport, client_id=conf.inverterid, keepalive=60, auth=conf.pubauth)
-                if conf.verbose: print("\t - " + 'MQTT message message sent') 
+                mqtttopic = conf.mqtttopic + '/' + definedkey["pvserial"]
+                publish.single(mqtttopic, payload=jsonmsg, qos=0, retain=True, hostname=conf.mqttip,port=conf.mqttport, client_id=conf.inverterid, keepalive=60, auth=conf.pubauth)
+                if conf.verbose: print("\t - " + 'MQTT message message sent on topic:', mqtttopic) 
             except TimeoutError:     
                 if conf.verbose: print("\t - " + 'MQTT connection time out error') 
             except ConnectionRefusedError:     
